polaris-react/src/components/TextField/TextField.tsx

```diff
@@ -160,6 +160,8 @@ interface NonMutuallyExclusiveProps {
   requiredIndicator?: boolean;
   /** Indicates whether or not a monospaced font should be used */
   monospaced?: boolean;
+  /** Indicates whether or not to blur the input and trigger any blur callbacks when the enter key is pressed */
+  blurOnEnterKeyPress?: boolean;
```

Comment:
```suggestion
/** Callback fired on key press */
  onKeyPress?(event: React.KeyboardEvent): void;
```

This would be best as a general callback. You can programmatically set `focused` to false or do whatever else you need to if the key pressed is Enter using the callback.

Reply:
[`onKeyPress` is deprecated ](https://developer.mozilla.org/en-US/docs/Web/API/Element/keypress_event) 😅